### PR TITLE
Add GitHub workflow to check for broken links

### PR DESCRIPTION
**Pull request by t1m0thyj**
For every pull request opened against the docs-staging or master branch, the workflow will run and check for broken links.

It will fail the build if any files changed in that pull request contain broken links.

It will also generate a report for all broken links found in MD files in the "docs" folder.

To view the report for this PR (or any other one in the future once this is merged):
1. Click the "Details" link for the "Link Checker" build
image
2. Click the "Summary" link in the sidebar:
image
3. Scroll down the page to see the list:
image

### Your checklist for this pull request
🚨Please review the [guidelines for contributing](../docs/contribute/contributing.md) to this repository.

- [ ] If the changes in this PR is part of the next future release, make this pull request against the **docs-staging** branch which will be published at the next release boundary. If the changes in this PR are part of the current release, use the default base branch, **master**. For more information about branches, see https://github.com/zowe/docs-site/tree/master#understanding-the-doc-branches. 
      
- [ ] If this PR relates to GitHub issues in `docs-site` or other repositories, please list in Description, prefixed with **close**, **fix** or **resolve** keywords.

### Description (including links to related git issues)
Please describe your pull request.

:heart:Thank you!

### CODE REVIEW

File: .github/workflows/link-checker.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        run: cat lychee/out.md | grep "# Errors in" | cut -f4- -d' ' > lychee/files.txt
+
+      - name: List changed files with errors
+        id: changed-files
+        uses: tj-actions/changed-files@v35
+        with:
+          files_from_source_file: lychee/files.txt
+
+      - name: Fail if new errors found
+        if: steps.changed-files.outputs.any_changed == 'true'
```
Comment (zFernand0) [inline .github/workflows/link-checker.yml:55]: Curious if this fails when reducing the number of broken links? 😋 
say that someone only addresses 1 broken link... 
will this consider the `changed-files.outputs.any_changed == true` ?

**Review by zFernand0 — APPROVED** [review]
Thanks for adding this.
I believe this will benefit the Docs Squad by reducing the number of broken links introduced.

LGTM! 😋 

We should probably wait for the Docs squad  lead to review this as well

File: .github/workflows/link-checker.yml (hunk 1 above)
Comment (t1m0thyj) [reply .github/workflows/link-checker.yml]: After consulting with the docs squad, decided to simplify the logic so the PR fails for any broken links found in the entire repo.

**Review by zFernand0 — APPROVED** [review]

**Comment by balhar-jakub** [comment]
Yeah, this looks like a great addition.

**Review by anaxceron — APPROVED** [review]
Thanks for doing this @t1m0thyj  -- everything looks good!

**Pull request merged**